### DISCUSSION TITLE
GitLense running extremely slow

### DISCUSSION THREAD

**Discussion by tomasvorel2** [gitkraken/vscode-gitlens#2769] [Q&A]
Hi eveyrone,

Does anyone else is having issues with the speed of the VS Code extension? When I open up a project, everything takes sooo long to load. Scanning for files, staging, commiting, stashing. I'm trying to figure out if it is a software or hardware issue (my computer not being fast enough). I'm using a Macbook Pro from 2018 (2,4 GHz Quad-Core Intel Core i5, 16GB of RAM) so it shouldn't be that slow right? My project is a Next.js site (about 1200 pages).

Any tips/pointers?

EDIT: I'm using a GitLens v14.0.1

Thanks!

**Comment by jkelroy** [comment]
Hello @tomasvorel2, to confirm, you are not seeing slow loading times in VS Code if you disable GitLens?

<img width="694" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/27739784/264419df-ba87-4197-8480-beee2232578e">


If the slowness does not persist with GitLens disabled, is this only happening with a specific repository or any? A quick check to make sure the local repository is not contributing to this is by taking a fresh clone of the repository in a new location.

If this persists with a fresh clone, by chance is this a public repository that can be shared for testing? If not, can you please enable debug logging by running from _GitLens: Enable Debug Logging_ from the Command Palette (<kbd>F1</kbd> or <kbd>ctrl/cmd</kbd>+<kbd>shift</kbd>+<kbd>p</kbd>). This will enable logging to the _GitLens_ and _GitLens (Git)_ channel in the _Output_ pane.  Once enabled, please try some actions in GitLens that are slow, and attach the log from the both _GitLens_ and _GitLens (Git)_ channels.
